docs: update CLA info (#15370)
eslint/eslint: CONTRIBUTING.md

## Contributing Code

> In order to submit code or documentation to an ESLint project, please sign our [Contributor License Agreement](https://eslint.org/docs/developer-guide/contributing/#signing-the-cla) and read over the [Pull Request Guidelines](https://eslint.org/docs/developer-guide/contributing/pull-requests).

Reviewer: I think we need a better place to link to. Right now the CLA links to a paragraph that says to sign the CLA, which in turn links to a GitHub repo where you can’t sign the CLA.

Reviewer 2: You can sign CLA there by making a test PR if you are not covered it will provide a link to sign the CLA process.

Reviewer 2: But I agree that if there was a valid CLA signing link which doesn't require sending a PR, then thats a great improvement.

Author: [This is the direct link](https://contributor.easycla.lfx.linuxfoundation.org/#/cla/project/3d9a9c15-cb4a-478d-a185-d8d2881dd914/user/f5e71bf5-4dd3-4e03-b70f-7459c4deb7df)

```suggestion
In order to submit code or documentation to an ESLint project, please sign our [Contributor License Agreement](https://contributor.easycla.lfx.linuxfoundation.org/#/cla/project/3d9a9c15-cb4a-478d-a185-d8d2881dd914/user/f5e71bf5-4dd3-4e03-b70f-7459c4deb7df) and read over the [Pull Request Guidelines](https://eslint.org/docs/developer-guide/contributing/pull-requests).
```

Reviewer: That one is tied to your name. :-/ Let me see if I can figure it out.

> In order to submit code or documentation to an ESLint project, you’ll be asked to sign our CLA when you send your first pull request but you can preview the CLA at this [URL](https://cla.openjsf.org/). Also, please read over the [Pull Request Guidelines](https://eslint.org/docs/developer-guide/contributing/pull-requests).

Reviewer: ```suggestion
In order to submit code or documentation to an ESLint project, you’ll be asked to sign our CLA when you send your first pull request.  (Read more about the Open JS Foundation CLA process at https://cla.openjsf.org/.) Also, please read over the [Pull Request Guidelines](https://eslint.org/docs/developer-guide/contributing/pull-requests).
```

> In order to submit code or documentation to an ESLint project, you’ll be asked to sign our CLA when you send your first pull request. (Read more about the Open JS Foundation CLA process at https://cla.openjsf.org/.) Also, please read over the [Pull Request Guidelines](https://eslint.org/docs/developer-guide/contributing/pull-requests).

Author: ```suggestion
In order to submit code or documentation to an ESLint project, you’ll be asked to sign our CLA when you send your first pull request. (Read more about the Open JS Foundation CLA process at <https://cla.openjsf.org/>.) Also, please read over the [Pull Request Guidelines](https://eslint.org/docs/developer-guide/contributing/pull-requests).
```
